presto-main/src/main/java/com/facebook/presto/execution/scheduler/SimpleNodeSelector.java

```diff
@@ -168,6 +172,23 @@ else if (!splitWaitingForAnyNode) {
         return new SplitPlacementResult(blocked, assignment);
     }
 
+    private ResettableRandomizedIterator<InternalNode> getRandomCandidates(int limit, NodeMap nodeMap, List<RemoteTask> existingTasks)
+    {
+        List<InternalNode> existingNodes = existingTasks.stream()
+                .map(remoteTask -> nodeMap.getNodesByNodeId().get(remoteTask.getNodeId()))
+                .collect(toList());
+
+        int alreadySelectedNodeCount = existingTasks.size();
+        int nodeCount = nodeMap.getNodesByNodeId().size();
+
+        if (alreadySelectedNodeCount < limit && alreadySelectedNodeCount < nodeCount) {
+            List<InternalNode> moreNodes =
+                    selectNodes(limit - alreadySelectedNodeCount, randomizedNodes(nodeMap, includeCoordinator, Sets.newHashSet(existingNodes)));
```

Comment:
We prefer immutable collections whenever possible. Please use `ImmutableSet.copyOf()`.

See the second bullet point here: https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development